packages/itwinui-css/src/progress-indicator/progress-indicator.scss

```diff
@@ -1,28 +1,200 @@
+    :is(svg, img) {
+      @include mixins.iui-icon-style('l');
+    }
   }
 
-  &:not(.iui-negative):not(.iui-positive) > .iui-track > .iui-indeterminate {
-    @include linear.iui-progress-indicator-linear-indeterminate;
+  &:where(:not([data-iui-indeterminate='true'])) {
+    &::after {
+      $percentage: var(--iui-progress-percentage, 100%);
+      mask-image: conic-gradient(#000 $percentage, #0000 $percentage), linear-gradient(#000 0 0);
+    }
+  }
+
+  &:where([data-iui-indeterminate='true']) {
+    &::after {
+      $_keyframe-name: iui-#{string.unique-id()};
+      mask-image: conic-gradient(#0000, #000), linear-gradient(#000 0 0);
+      animation: $_keyframe-name 0.8s cubic-bezier(0.6, 0.4, 0.4, 0.6) infinite;
+
+      @keyframes #{$_keyframe-name} {
+        100% {
+          transform: rotate(360deg);
+        }
+      }
+    }
   }
 }
 
-.iui-progress-indicator-radial {
-  @include radial.iui-progress-indicator-radial;
+// ----------------------------------------------------------------------------
+
+.iui-progress-indicator-linear-label {
+  display: flex;
+  align-items: center;
+  font-size: var(--iui-font-size-0);
+  justify-content: space-between;
+  color: var(--_iui-progress-indicator-linear-label-color);
+  --iui-svg-fill: var(--_iui-progress-indicator-linear-label-fill);
+
+  > :only-child {
+    margin-inline: auto;
+  }
+}
 
-  &.iui-determinate {
-    @include radial.iui-progress-indicator-radial-determinate;
+.iui-progress-indicator-linear {
+  display: grid;
+  gap: var(--iui-size-2xs);
+
+  &::before {
+    content: '';
+    block-size: var(--iui-size-2xs);
+    $percentage: var(--iui-progress-percentage, 100%);
+    $fill: var(--_iui-progress-indicator-track-fill, var(--iui-color-border-accent));
+
+    background-color: var(--iui-color-border);
+    background-image: linear-gradient(90deg, $fill 0% 100%);
+    background-repeat: no-repeat;
+    background-size: $percentage 100%;
+    forced-color-adjust: none;
+
+    @media (forced-colors: active) {
+      border: solid 0.5px;
+      background-color: transparent;
+      background-image: linear-gradient(90deg, CanvasText 0% 100%);
+    }
+  }
+
+  @each $status in positive, negative {
+    &[data-iui-status='#{$status}'] {
+      --_iui-progress-indicator-track-fill: #{var(--iui-color-border-#{$status})};
+      --_iui-progress-indicator-linear-label-color: var(--iui-color-text-#{$status});
+      --_iui-progress-indicator-linear-label-fill: var(--iui-color-icon-#{$status});
+      @include mixins.iui-text-selection($status);
+    }
   }
 
-  &.iui-indeterminate {
-    @include radial.iui-progress-indicator-radial-indeterminate;
+  &:where([data-iui-animated='true']:not([data-iui-indeterminate='true'])) {
+    &::before {
+      transition: background-size 1s ease-in-out;
+    }
+  }
+
+  &:where([data-iui-indeterminate='true']) {
+    &::before {
+      $_keyframe-name: iui-#{string.unique-id()};
+      animation: $_keyframe-name 1.5s infinite linear;
```

Comment:
Maybe we could use same cubic-bezier function as in radial?

Reply:
maybe. the difference is that the keyframes are already weirdly spaced out (0%, 40%, 100%, with different sizes in each) so i tried to keep it similar to how it felt before

if you want to play with the easing function and keyframes, feel free to suggest something specific

Reply:
@gretanausedaite lmk if you have a specific suggestion for changing this animation or if we can merge this as-is

Reply:
Merge as is.